edge-modules/azure-monitor/src/IotHubUpload/IotHubUpload.cs

```diff
@@ -48,6 +57,197 @@ public async Task<bool> PublishAsync(IEnumerable<Metric> metrics, CancellationTo
             }
         }
 
+        private string Transform(IEnumerable<ExportMetric> metrics)
```

Comment:
You are looking for very specific fields in the metric. What's happening when we add a new metric with different fields?